webapp/src/pages/Settings.tsx

```diff
@@ -31,6 +45,20 @@ const Settings: React.FC = () => {
 
   const resultingConfig = { ...data, ...partialConfig };
```

Comment:
I suspect this would also need `_.merge()`.
```suggestion
  const resultingConfig = _.merge(data, partialConfig);
```